python/lsst/pipe/base/_task_metadata.py

```diff
@@ -477,6 +516,46 @@ def __delitem__(self, key: str) -> None:
             # Report the correct key.
             raise KeyError(f"'{key}' not found'") from None
 
+    def get_dict(self, key: str) -> NestedMetadataDict:
+        """Return a possibly-hierarchical nested `dict`.
+
+        This implements the `GetDictMetadata` protocol for consistency with
+        `lsst.daf.base.PropertySet` and `lsst.daf.base.PropertyList`.
+
+        Parameters
+        ----------
+        key : `str`
+            String key associated with the mapping.
+
+        Returns
+        -------
+        value : `~collections.abc.Mapping`
+            Possibly-nested mapping, with `str` keys and values that are `int`,
+            `float`, `str`, `bool`, or another `dict` with the same key and
+            value types.  Will be empty if ``key`` does not exist.
```

Comment:
Maybe mention this is not a view and so changes to the returned dict won't affect the original?